fhir-codegen/gradle-plugin/src/main/kotlin/com/google/fhir/codegen/FhirCodegen.kt

```diff
@@ -59,13 +64,14 @@ object FhirCodegen {
       mutableListOf(
         modelFileSpec
           .addType(
-            ModelTypeSpecGenerator.generate(
-              modelClassName,
-              structureDefinition,
-              isBaseClass,
-              surrogateFileSpec,
-              serializerFileSpec,
-            )
+            ModelTypeSpecGenerator(valueSetMap, codeSystemMap, commonBindingValueSetUrls)
```

Comment:
so we create this generator for each call to the `generateFileSpecs` function? shouldn't we create this generator in the codegen task file, and just pass that?

Reply:
I do not fully understand your request here. Please clarify.